### DISCUSSION TITLE
Can we use this framework for existing Users database ?

### DISCUSSION THREAD

**Discussion by leelu** [fief-dev/fief#183] [Q&A]
Hi 

Can we use Fief for an existing User database with regular email/password login ? Also does it support Apple Login ?

Thanks
Leela

**Comment by fief-bailiff** [comment]
Hail, @leelu 👋 Welcome to Fief's kingdom!

Our team will get back to you very soon to help.

In the meantime, take a minute to star our repository ⭐️

![star-fief](https://user-images.githubusercontent.com/1144727/192523552-dc2c31fa-ae94-42fa-9ab4-7d0ba51bb7b0.png)

Farewell!

**Comment by frankie567** [comment]
Currently, there is no mechanism to import existing data to the database. However, if you use BYOD or self-host, it might be possible to import directly your data in the SQL database. You need to conform to the [schema of the `fief_users` table](https://github.com/fief-dev/fief/blob/c7b99d65a64f187d99ada818848133f9a497ec0d/fief/models/user.py#L28-L42). Password hashes might be imported as well if they are hashed using the [argon2 or bcrypt algorithms](https://github.com/fief-dev/fief/blob/main/fief/crypto/password.py).

Apple Login is currently not supported but it's on our roadmap!
